src/omdb_data_merge.sql

```diff
@@ -134,4 +99,116 @@ INSERT INTO media_production_company (media_id, production_company_id, type)
 SELECT m.media_id, p.production_company_id, m.type
 FROM media m
 JOIN prod_comps o ON o.tconst = m.imdb_id
-JOIN production_company p ON p.name = o.comp;
\ No newline at end of file
+JOIN production_company p ON p.name = o.comp;
+
+
+--SCORE 
+-- Using ratings from omdb_data as it has rottentomatoes
+-- imdb score and metacatscore in omdb_data are the same
+
+--extracting data from JSON 
+--Changing source names
+--Fixing values
+--collecting vote counts for imdb
+WITH json_extract AS (
+    SELECT
+        o.tconst,
+        m.media_id,
+        CASE
+            WHEN json_data->>'Source' = 'Internet Movie Database' THEN 'IMDb'
+            WHEN json_data->>'Source' = 'Rotten Tomatoes' THEN 'RT'
+            WHEN json_data->>'Source' = 'Metacritic' THEN 'MC'
+            ELSE NULL
+        END AS source,
+        CASE
+            WHEN json_data->>'Source' = 'Internet Movie Database' 
+                THEN REPLACE(json_data->>'Value', '/10', '')
+            WHEN json_data->>'Source' = 'Rotten Tomatoes'
+                THEN REPLACE(json_data->>'Value', '%', '')
+            WHEN json_data->>'Source' = 'Metacritic'
+                THEN REPLACE(json_data->>'Value', '/100', '')
+            ELSE NULL
+        END AS value,
+        CASE
+            WHEN o.imdbvotes = 'N/A' OR o.imdbvotes = '' THEN 0
+            WHEN json_data->>'Source' = 'Internet Movie Database'
+                THEN REPLACE(o.imdbvotes, ',', '')::INTEGER
+            WHEN jsonb_array_length(o.ratings::jsonb) = 0
+                AND (o.imdbvotes != 'N/A' OR o.imdbvotes != '')
+                THEN REPLACE(o.imdbvotes,',','')::INTEGER
+            ELSE 0
+        END AS imdbvotes
+    FROM original.omdb_data o
+    JOIN media m ON m.imdb_id = o.tconst
+    LEFT JOIN LATERAL jsonb_array_elements(o.ratings::jsonb) AS json_data ON TRUE
+),  
```

Comment:
Not sure I fully understand what is going on here but looks cool 😅

Reply:
So i made this dummy table. For storing the data from rating column. 
As it is json array which had objects with value and source.
>> LEFT JOIN LATERAL jsonb_array_elements(o.ratings::jsonb) AS json_data ON TRUE
o.ratings::jsonb - converts string to json
jsonb_array_elements()splits the json array into separate objects

So the code creates separate rows for the elements in array. 
So I got the data for imdb, RT and MC in different rows of my dummy table. 

Then I used CASE to fix the data as it was in bad format. 
>>json_data->>'Source' 
This extracts the value from json object variable Source
and then just bunch of conditions and fixes as imdbvotes was string with comas between tousands
converting to integer after

And then I just used that table with corrected and extracted values for update and insert